File: src/main/java/com/rtkit/upsource_manager/config/AdminConfig.java
package com.rtkit.upsource_manager.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;

@Configuration
@PropertySource("classpath:auth.properties")
public class AdminConfig {
    public static String ADMIN_BASIC_AUTH;

    @Value(value = "${auth.data}")
    public void setAdminBasicAuth(String data) {
        ADMIN_BASIC_AUTH = "Basic " + data;
    }

}

File: src/main/java/com/rtkit/upsource_manager/config/_AdminConfig.kt
package com.rtkit.upsource_manager.config

import org.springframework.context.annotation.Configuration

/**
 * Шаблон для создания класса AdminConfig
 *
 */
@Configuration
open class _AdminConfig {
    companion object {
        /**
         * В этой переменной хранится логин и пароль админа, от которого будут выполняться большинство запросов
         * Кодировка: Base64
         * Формат: login:password
         */
        private const val encodedData: String = ""
        const val ADMIN_BASIC_AUTH = "Basic $encodedData"
    }
}